gdal/doc/source/user/sql_sqlite_dialect.rst

```diff
@@ -21,6 +21,11 @@ update, INSERT/UPDATE/DELETE statements can also be run. GDAL is internally usin
 `the Virtual Table Mechanism of SQLite <https://sqlite.org/vtab.html>`_
 and therefore operations like ALTER TABLE are not supported.
 
+If the datasource is SQLite database (GeoPackage, SpatiaLite) then SQLite dialect
+acts as native SQL dialect and Virtual Table Mechanism is not used. It is possible to
+force GDAL to use Virtual Tables even in this case by specifying 
+"-dialect INDIRECT_SQLITE".
```

Comment:
Perhaps with this additional note:
```suggestion
"-dialect INDIRECT_SQLITE". This should be used only when necessary, since going through the virtual table mechanism might affect performance.
```